crates/edr_napi/src/solidity_tests/test_suite.rs

```diff
@@ -0,0 +1,100 @@
+use foundry_compilers::artifacts::Libraries;
+use napi_derive::napi;
+
+/// A test suite is a contract and its test methods.
+#[derive(Clone)]
+#[napi(object)]
+pub struct TestSuite {
+    /// The identifier of the test suite.
+    pub id: ArtifactId,
+    /// The test contract.
+    pub contract: TestContract,
+}
+
+/// The identifier of a Solidity test contract.
+#[derive(Clone)]
+#[napi(object)]
+pub struct ArtifactId {
+    /// The name of the contract.
+    pub name: String,
+    /// Original source file path.
+    pub source: String,
+    /// The solc semver string.
+    pub solc_version: String,
+    /// The artifact cache path. Currently unused.
+    pub artifact_cache_path: String,
+}
+
+impl TryFrom<ArtifactId> for foundry_compilers::ArtifactId {
+    type Error = napi::Error;
+
+    fn try_from(value: ArtifactId) -> napi::Result<Self> {
+        Ok(foundry_compilers::ArtifactId {
+            path: value.artifact_cache_path.parse().map_err(|_err| {
+                napi::Error::new(napi::Status::GenericFailure, "Invalid artifact cache path")
+            })?,
+            name: value.name,
+            source: value.source.parse().map_err(|_err| {
+                napi::Error::new(napi::Status::GenericFailure, "Invalid source path")
+            })?,
+            version: value.solc_version.parse().map_err(|_err| {
+                napi::Error::new(napi::Status::GenericFailure, "Invalid solc semver string")
+            })?,
+        })
+    }
+}
+
+/// A test contract to execute.
+#[derive(Clone)]
+#[napi(object)]
+pub struct TestContract {
+    /// The contract ABI as a JSON string.
+    pub abi: String,
```

Comment:
I'm writing this comment here, as this reminded me of it.

Hardhat uses a non-standard ABI for `console.log`. This might have unexpected consequences if we're compiling through Hardhat while using parts of `foundry` that make have expectations of the ABI.

Raising this for awareness, as it might have some impact on tests failing unexpectedly

cc: @fvictorio